alas_but_one.py

```diff
@@ -1,39 +1,108 @@
 import os
 import re
-from collections import Counter
+from spellchecker import SpellChecker
+from collections import defaultdict
+import string
+import argparse
 
 # Usage
 # -----
 # This script reads files in the relative "source/" directory and sub-directories
-# named with the "rst" or "txt" extension and counts the instances of each
-# word (as matched by the WORD_REGEX constant).
-# Update the MAX_OCCUR=n constant to print out only words that are repeated
-# up to n times.
+# named with the "rst" or "txt" extension and outputs a list of words that occur
+# only at a frequency that you specify. By default, the script runs a
+# spellchecker on the output list and separates the list
+# into words that are spelled correctly and words that are potentially misspelled.
 #
 # Backstory
 # ---------
-# A user reported an issue with documentation in which "Atlas" was misspelled 
-# as "Alas". After investigation, a writer reported there was only one instance 
+# A user reported an issue with documentation in which "Atlas" was misspelled
+# as "Alas". After investigation, a writer reported there was only one instance
 # of the type. Hence, "there was, alas, but one".
+#
+# Instructions
+# ------------
+# Run this file at the same level as your "/source" directory.
+# Update the OCCUR_FREQ=n constant in the main method to locate only words that are repeated
+# up to n times.
+# When you run the file, you can include the optional "--noSpellCheck"
+# flag to disable the spellchecker function:
+# e.g. python3 alas_but_one.py --noSpellCheck > output.txt
+
+
+def count_in_dir():
+    directory = "source"
+    filename_re = r"\.(txt|rst)$"
+    pattern = re.compile(r"\s*(?:[()]|'s|\s|-|`|/|#)\s*")
+
+    processed = []
+    words_and_freqs = []
+
+    for root, dirs, files in os.walk(directory):
+        for filename in files:
+            if re.search(filename_re, filename):
+                filepath = os.path.join(root, filename)
+                # print(filepath)
+                read_file = open(filepath, "r", encoding="utf-8")
+                # processed = []
+
+                for line in read_file:
+                    line = line.strip()
+                    words = pattern.split(line)
+
+                    for word in words:
+                        pword = word.strip(string.punctuation)
+                        processed.append(pword.lower())
```

Comment:
Suggestion:

I think ``lower()`` in the original script was a mistake and that it should not be included as it can generate false negatives. E.g. "cOMPASS" should not be counted the same as "Compass".